modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java

```diff
@@ -1905,10 +1905,13 @@ public Parameter getParameter(ObjectNode obj, String location, ParseResult resul
 				parameter = new Parameter();
 				String mungedRef = mungedRef(ref.textValue());
 				if (mungedRef != null) {
-					parameter.set$ref(mungedRef);
-				} else {
-					parameter.set$ref(ref.textValue());
-				}
+                    parameter.set$ref(mungedRef);
+                }else {
+                    parameter.set$ref(ref.textValue());
+                }
+                if(ref.textValue().startsWith("#/components") && !(ref.textValue().startsWith("#/components/parameters")||ref.textValue().startsWith("#/components/headers"))) {
```

Comment:
Why is it accepting a ref to `#/components/headers` for parameters? (and the other way around below?)

Reply:
headers and parameters do share the same structure, and they can use with one or the other.
Docs: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.0.md#header-object
tests that use it: https://github.com/swagger-api/swagger-parser/blob/707adcb2fe3a2d31e8c915d9e4cacdb3b176f7c2/modules/swagger-parser-v3/src/test/java/io/swagger/v3/parser/test/OpenAPIV3ParserTest.java#L524